generators/schema/index.js

```diff
@@ -0,0 +1,239 @@
+'use strict';
+
+var yeoman = require('yeoman-generator');
+var extend = require('util')._extend;
+var path = require('path');
+var chalk = require('chalk');
+var glob = require('glob');
+var utils = require('./../utilities');
+
+var requiredResources = [];
+
+// TODO: We should get these lists from the cloud.
+//       Temporarily getting the list from a locally synced repo.
+function getKnownResources(resourceChoices, resourceInfos, repoRoot) {
+  var paths = glob.sync(repoRoot + '/oic.r.*/', {});
+
+  paths.forEach(function (resourcePath) {
+    var ramlFiles = glob.sync(resourcePath + '*.raml', {});
+    if (ramlFiles.length > 0) {
+      var resourceName = path.basename(ramlFiles[0], '.raml');
+      var resourceSchema = path.basename(resourcePath);
+      resourceName = resourceName.charAt(0).toUpperCase() + resourceName.slice(1);
+      resourceInfos.push({ schema: resourceSchema, name: resourceName });
+      resourceChoices.push(resourceSchema);
+    }
+  });
+}
+
+function getResourceInfo(resourceSchema, resourceInfos) {
+  for (var i = 0; i < resourceInfos.length; i++) {
+    if (resourceInfos[i].schema === resourceSchema) {
+      return resourceInfos[i];
+    }
+  }
+
+  return undefined;
+}
+
+function addRequiredResource(resourceInfo) {
+  if (requiredResources.indexOf(resourceInfo) === -1) {
+    requiredResources.push(resourceInfo);
+  }
+}
+
+module.exports = yeoman.Base.extend({
+  constructor: function () {
+    yeoman.Base.apply(this, arguments);
+
+    this.option('repo-root', {
+      desc: 'The root of the translators repo',
+      alias: 'r',
+      type: String
+    });
+
+    this.props = { devices: [] };
+
+    var resourceChoices = [];
+    var resourceInfos = [];
+    var repoRoot;
+
+    if (this.options['repo-root'] === undefined) {
+      this.log(chalk.bold.red('WARNING: ') + 'No repo specified. Defaulting to ../translators/');
+      repoRoot = '../translators/';
+    } else {
+      repoRoot = this.options['repo-root'];
+    }
+
+    getKnownResources(resourceChoices, resourceInfos, repoRoot);
+
+    this.addResource = function (device, message) {
+      var that = this;
+      var isFirst = device.resources.length === 0;
+      var shouldAdd = function (response) {
+        return isFirst || response.addResource;
+      };
+
+      var prompts = [
+        {
+          when: shouldAdd,
+          type: 'rawlist',
+          name: 'resourceSchema',
+          message: isFirst ? 'What schema does the first resource use?' : 'What schema does the resource use?',
+          choices: resourceChoices
+        },
+        {
+          when: shouldAdd,
+          type: 'input',
+          name: 'resourceName',
+          message: 'What is the resource name (e.g. colorMode)?',
```

Comment:
>color [](start = 52, length = 5)

Nit: "Colour" (looks like OCF prefers the British variant instead of NorthAmerican)